src/main/java/com/cs56fitnessapp/models/workout/StrengthTraining.java

```diff
@@ -0,0 +1,90 @@
+
+import com.cs56fitnessapp.models.User;
+import java.time.LocalDate;
+
+/**
+ * @author Daniel Cervantes
+ * @author Ksenia Koldaeva
+ * Date created: 10/18/17
+ * Last updated: 11/11/17
+ */
+
+public class StrengthTraining extends Workout {
+
+    private StrengthTrainingLevel strengthTrainingLevel;
+    private boolean training;
+
+    /**
+     * Constructs Strength Training object with provided parameters
+     * @param user user performing Strength Training
+     * @param date date of Strength Training
+     * @param timePerformingHours amount of time of Strength Training in hours
+     * @param strengthTrainingLevel one of three different kinds of strength training levels
+     */
+
+    public StrengthTraining(User user, LocalDate date, double timePerformingHours, StrengthTrainingLevel strengthTrainingLevel) {
+        super(user, date);
+        this.setTimePerformingHours(timePerformingHours);
+        this.strengthTrainingLevel = strengthTrainingLevel;
+
+    }
+
+    // Setters and Getters
+    public StrengthTrainingLevel getStrengthTrainingLevel() {
+        return strengthTrainingLevel;
+    }
+
+    public void setStrengthTrainingLevel(StrengthTrainingLevel strengthTrainingLevel) {
+        this.strengthTrainingLevel = strengthTrainingLevel;
+    }
+
+    public boolean isTraining() {
+        return training;
+    }
+
+    public void setTraining(boolean training) {
+        this.training = training;
+    }
+
+
+    /**
+     * @return calories out from Strength Training based on
+     * the level of weights that were used during the workout
+     */
+
+    @Override
+    public int getCaloriesOut() {
+        int caloriesOut = 0;
+
+        switch (strengthTrainingLevel) {
+            case HEAVY:
+                if (training) {
+                    caloriesOut = FitnessFormulas.caloriesOutByMET(FitnessFormulas.MET_STRENGTH_TRAINING_HEAVY, this.getUser().getbodyMassKg(), this.getTimePerformingHours());
+                }
+                break;
+            case MEDIUM:
+                if (training) {
+                    caloriesOut = FitnessFormulas.caloriesOutByMET(FitnessFormulas.MET_STRENGTH_TRAINING_MEDIUM, this.getUser().getbodyMassKg(), this.getTimePerformingHours());
+                }
+                break;
+            case LIGHT:
+                if (training) {
+                    caloriesOut = FitnessFormulas.caloriesOutByMET(FitnessFormulas.MET_STRENGTH_TRAINING_LIGHT, this.getUser().getbodyMassKg(), this.getTimePerformingHours());
+                }
+                break;
+        }
+        return caloriesOut;
+    }
+
+    /**
+     * Will be using getActiveTimeMins() to calculate the amount
+     * of minutes the user is active during the workout
+     * @return total active minutes from Strength Training
+     */
+    @Override
+    public int getActiveTimeMins(){
+        return (int)(this.getTimePerformingHours() - this.getCoolDownTimeHrs() - this.getWarmUpTimeHrs());
+    }
+
+}
```

Comment:
I think the initial logic was that `timePerforming` eventually the same as `activeTime`. `coolDownTime` and `warmUpTime` were optional params simply for additional information in the user cares about it.
For now, lets implement one method in Workout class (it is relevant for all types of Workout), like this: 

```java
    @Override
    public int getActiveTimeMins() {
        return (int)Math.round(FitnessFormulas.MINS_IN_AN_HOUR * this.getTimePerformingHours());
    }
```

I will also remove the same method from `Endurance` class.
It is pure design question and it can be finalized different ways - we can discuss on a meeting and then refactor.